### ISSUE TITLE
Native crash SIGSEGV: Segfault   __kernel_rt_sigreturn

### ISSUE THREAD

**Issue by ToniNikolaev23** [facebook/react-native#34745]
### Description

Hello everyone, today i saw in our Sentry that we have some unhandled issues. I will post screenshot from Sentry.

![Sentry](https://user-images.githubusercontent.com/72339606/191431050-236e5570-af9f-4782-9f37-7c489018d6bc.png)

### Version

0.64.3

### Output of `npx react-native info`

System:
    OS: macOS 12.6
    CPU: (10) arm64 Apple M1 Max
    Memory: 3.62 GB / 32.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 18.6.0 - /opt/homebrew/bin/node
    Yarn: 1.22.19 - /usr/local/bin/yarn
    npm: 8.13.2 - /opt/homebrew/bin/npm
    Watchman: 2022.07.04.00 - /opt/homebrew/bin/watchman
  Managers:
    CocoaPods: 1.11.3 - /opt/homebrew/bin/pod
  SDKs:
    iOS SDK:
      Platforms: DriverKit 21.4, iOS 15.5, macOS 12.3, tvOS 15.4, watchOS 8.5
    Android SDK: Not Found
  IDEs:
    Android Studio: 2021.2 AI-212.5712.43.2112.8609683
    Xcode: 13.4.1/13F100 - /usr/bin/xcodebuild
  Languages:
    Java: 11.0.15 - /usr/bin/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: 17.0.1 => 17.0.1
    react-native: 0.64.3 => 0.64.3
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found

### Steps to reproduce

I dont know how to reproduce.When i test with android studio, sometimes after launch app emulator just crash and 2nd launch is good. Maybe in real device is the same.

### Snack, code example, screenshot, or link to a repository

![Sentry](https://user-images.githubusercontent.com/72339606/191431348-2bcfa572-7ee0-4be0-88e6-b1e2d0fef514.png)

**Comment by nnabinh** [comment]
Same problem

![Screen Shot 2022-09-30 at 12 27 08](https://user-images.githubusercontent.com/10094591/193196702-5512041e-6e30-4ea7-8ca3-b652538f34b0.png)
![Screen Shot 2022-09-30 at 12 27 23](https://user-images.githubusercontent.com/10094591/193196706-b7457389-08eb-4291-90b0-9395179ddb8e.png)

**Comment by cortinico** [comment]
There is nothing we can without a reproducer sadly.
Are you able to reproduce locally in any form?

**Comment by jay-geocloud** [comment]
I also have been noticing these __kernel_rt_sigreturn  Sentry errors for a few months now but not sure what's the issue at the moment. 

<img width="1112" alt="Screen Shot 2022-10-03 at 4 16 02 PM" src="https://user-images.githubusercontent.com/71744965/193671794-9b6f45d2-063a-418c-89e8-01673126d290.png">
<img width="1079" alt="Screen Shot 2022-10-03 at 4 15 35 PM" src="https://user-images.githubusercontent.com/71744965/193671809-3cd002ff-586a-4a37-932b-d1cfdb77a793.png">

**Comment by BatDroid** [comment]
after trying a lot, I removed the `expo-sentry` (`Sentry.init`) and I couldn't reproduce it on my Samsung device (as this happens there) any more. so seeing this issue on a `react-native` channel, I suppose this issue should be related to `react-native-sentry` rather than `expo-sentry`

**Comment by cortinico** [comment]
> I suppose this issue should be related to `react-native-sentry` rather than `expo-sentry`

Don't you mind opening an issue here https://github.com/getsentry/sentry-react-native/issues and link against this one?

**Comment by BatDroid** [comment]
@cortinico I'm sorry for the wrong information I gave (didn't do it intentionally). As a matter of the fact, it wasn't `expo-sentry`, I just removed my `yarn.lock` and installed packages again. 
I know how crazy this can be, but that was the only way for my project :)

**Comment by kumarparth380** [comment]
I got this issue fixed by enabling Hermes. Earlier, hermes in my project was disabled because of some reason so had to turn it on to avoid these crashes. Now the app works fine on Android 12.

**Comment by trungls1706** [comment]
i got same problem, in samsung device

**Comment by ys-sherzad** [comment]
I'm also noticing quite a number of this error. But don't have Hermes enabled and I'm wondering if the issue was solved for you @kumarparth380 !

**Comment by nlarif** [comment]
Did not see this error again since I switched to Hermes on Android: https://docs.expo.dev/guides/using-hermes/#switch-javascript-engine-on-a-specific-platform

**Comment by Carl-MONNERA** [comment]
👋🏻 Same here, hermes is disabled (Galaxy S21+ 5G Android 13)

**Comment by mitchiemt11** [comment]
I'm still having this issue on Android 13 RN v0.67.5. Is there a possible solution @ToniNikolaev23 ?? I have enabled Hermes but nothing works.

**Comment by Joshandrews43** [comment]
Also getting this crash as well

**Comment by RalissonMattias** [comment]
Here it also started showing up heavily in sentry and crashlytics, my app crashes sometimes randomly. It started to appear after RN 0.71.*

**Comment by MianSaad705** [comment]
@RalissonMattias  Are you using react-native-screens, if yes then which version secondly you are able to reproduce this crash locally?

**Comment by RalissonMattias** [comment]
Yes, I'm using `react-native-screens: 3.19.0` and the crash doesn't happen locally

**Comment by MianSaad705** [comment]
@RalissonMattias according to my understanding it's a crash on a fresh install and occur only one time, facing the same issue.

**Comment by RalissonMattias** [comment]
@MianSaad705 Yes, the same thing happens here. In your application hermes is true or false?

**Comment by MianSaad705** [comment]
@RalissonMattias  hermes is false in my app.

**Comment by RalissonMattias** [comment]
@MianSaad705 That must be it then, in my tests with hermes turned off, the app always closes the first time I download and enter the app. With enabled hermes this is not happening. I used to leave hermes enabled in my app, but it caused a performance drop in some situations, with it disabled the app works better

**Comment by ys-sherzad** [comment]
I can confirm that after enabling **Hermes** on Android, we're no longer getting this crash.

**Comment by jefflewis** [comment]
I've commented on what appears to be linked issues:

https://github.com/expo/expo/issues/18083
https://github.com/facebook/react-native/issues/33896

I see this same behavior with samsung devices on first launch with JSC. I saw this when migrating to RN 0.70.6 (and persisted when trying to go to latest 0.71.3).

```
adb logcat | ~/Android/sdk/ndk/23.1.7779620/ndk-stack -sym jni/arm64-v8a/libjsc.so
********** Crash dump: **********
Build fingerprint: 'samsung/z3quew/z3q:12/SP1A.210812.016/G988U1UES3FVK1:user/release-keys'
#00 0x00000000005032e4 /data/app/~~VtVNBRifuJ04N56RIsWoRQ==/my.app.bundle.id--LnOREf5niSbPh06Kw3gXw==/base.apk!libjsc.so (BuildId: 9cb83e4124e3ee1471bc717c722b4109614385ba)
#01 0x0000000000000678 [vdso] (__kernel_rt_sigreturn+0)
#02 0x00000000000a30b8 /apex/com.android.runtime/lib64/bionic/libc.so (__epoll_pwait+8) (BuildId: b5d187ac75fe4580e13ab2ec624bf7f1)
#03 0x0000000000017ea8 /system/lib64/libutils.so (android::Looper::pollInner(int)+184) (BuildId: 748948a5650ad93d18b12eb1d9a51a89)
#04 0x0000000000017d84 /system/lib64/libutils.so (android::Looper::pollOnce(int, int*, int*, void**)+116) (BuildId: 748948a5650ad93d18b12eb1d9a51a89)
#05 0x0000000000159170 /system/lib64/libandroid_runtime.so (android::android_os_MessageQueue_nativePollOnce(_JNIEnv*, _jobject*, long, int)+48) (BuildId: 33b6cccb8cb818db0b8648d93b8de614)
#06 0x00000000003fe504 /data/misc/apexdata/com.android.art/dalvik-cache/arm64/boot.oat (art_jni_trampoline+116)
Crash dump is completed
```

**Comment by navarrofracisco** [comment]
Any updates on this, this is still happening as reported by our Sentry logs.
![image](https://user-images.githubusercontent.com/8353831/221272292-a400737f-07d1-4422-8093-4f7ef1adeeef.png)

**Comment by kkureli** [comment]
same, any solution?

**Comment by dhavalsulhuf** [comment]
I'm facing the same issue, any luck?

**Comment by Lakston** [comment]
We are having the same sentry reports, I haven't tried enabling Hermes yet, I'll test that soon

**Comment by ccdwyer** [comment]
> > I'm facing the same issue, any luck?
> 
> enabling hermes is solving issue

Enabling Hermes isn't an option for many of us, so long as it doesn't support Intl (and thus breaks dayjs-timezone and several other packages).

**Comment by raoufmadani** [comment]
> Disabling minification is solving the issue

Would you please clarify that ?

**Comment by ArthurJahn** [comment]
I'm facing the same issue. Any updates on this one?

**Comment by jefflewis** [comment]
This crash report/issue for jsc-android matches my earlier comment and the description of this bug. I commented on this issue that I managed to create a JS bundle that does and does not exhibit the crash. I'm at a loss for how to set up a test harness to actually begin debugging JSC to begin to patch the actual issue. 

I'm open to suggestions!

https://github.com/react-native-community/jsc-android-buildscripts/issues/180

**Comment by ShepSims** [comment]
Bump because this is still happening to me regularly... seeing about .1% of users crash on the first app load with this issue

**Comment by sergeushenecz** [comment]
Hi. I have the same issue on samsung devices.

**Comment by NawazKhanMd** [comment]
Same, this is giving me a headache.

**Comment by adityatamhane** [comment]
We are getting the same issue in crashanalytics

**Comment by miallo** [comment]
We definitley get this error on all Android versions including 14. We see it a lot Samsung, but also on all others (e.g. many different Pixels or Redmi).
Because we need to support old Android/iOS versions we are still on react-native@0.72 and since there is a performance issue in hermes (that was only recently solved) we can't enable hermes...

**Comment by omkarKiranaClub** [comment]
![Screenshot 2024-09-27 at 1 17 20 PM](https://github.com/user-attachments/assets/1ca8be38-07cc-43c7-935f-485486f3dabc)
<img width="306" alt="Screenshot 2024-09-26 at 2 57 33 PM" src="https://github.com/user-attachments/assets/ead652d3-9c1b-4760-aa36-f252b8da84d7">

I'm encountering persistent crashes in my React Native app (version 0.72.7) with Hermes enabled. I've analyzed the memory graph for a video component in my app, which has raised concerns about potential memory leaks. I'm seeking guidance on the following:

What is considered ideal or normal memory consumption for a React Native app?
Could the memory usage I'm observing be causing these crashes, possibly due to memory leaks?
Are there recommended tools or strategies for diagnosing and resolving memory-related issues in React Native apps?

I'd appreciate insights from the community on how to approach these performance issues and optimize my app's memory management. If additional information is needed, such as specific crash logs or detailed memory usage data, please let me know, and I'll provide them promptly.

**Comment by talaikis** [comment]
+ 1 RN 0.,75.5 new arch + hermes

**Comment by ignaciotcrespo** [comment]
Still happening here to 1% of users, RN 0.73 + hermes

**Comment by gokul1630** [comment]
Still, this issue occurred on a Samsung device
device: Samsung Galaxy Z Fold7 5G
Android version: 16
react-native: 0.78.3
New Arch & Hermes: enabled

```
SIGSEGV: NDK crash detected with signal: SIGSEGV

0  pc 00000848  [vdso] (__kernel_rt_sigreturn+0)
1  pc 006550a4  /system/lib64/libhwui.so
2  pc 004ad108  /system/lib64/libhwui.so
3  pc 004ac3bc  /system/lib64/libhwui.so
4  pc 0069da50  /system/lib64/libhwui.so
5  pc 004aae04  /system/lib64/libhwui.so
6  pc 006ddea4  /system/lib64/libhwui.so
7  pc 009157b4  /system/lib64/libhwui.so
8  pc 00924cac  /system/lib64/libhwui.so
9  pc 00720b4c  /system/lib64/libhwui.so
10  pc 00491100  /system/lib64/libhwui.so
11  pc 0079f18c  /system/lib64/libhwui.so
12  pc 0079ecc0  /system/lib64/libhwui.so
```
